perception/autoware_tensorrt_bevdet/src/bevdet.cpp

```diff
@@ -0,0 +1,717 @@
+  curr_e2g_transform(3, 3) = 1.f;
+  adj_e2g_transform(3, 3) = 1.f;
+
+  Eigen::Matrix4f currEgo2adjEgo = adj_e2g_transform.inverse() * curr_e2g_transform;
+  Eigen::Matrix3f currEgo2adjEgo_2d;
+  for (int i = 0; i < 2; i++) {
+    for (int j = 0; j < 2; j++) {
+      currEgo2adjEgo_2d(i, j) = currEgo2adjEgo(i, j);
+    }
+  }
+  currEgo2adjEgo_2d(2, 0) = 0.f;
+  currEgo2adjEgo_2d(2, 1) = 0.f;
+  currEgo2adjEgo_2d(2, 2) = 1.f;
+  currEgo2adjEgo_2d(0, 2) = currEgo2adjEgo(0, 3);
+  currEgo2adjEgo_2d(1, 2) = currEgo2adjEgo(1, 3);
+
+  Eigen::Matrix3f gridbev2egobev;
+  gridbev2egobev(0, 0) = x_step_;
+  gridbev2egobev(1, 1) = y_step_;
+  gridbev2egobev(0, 2) = x_start_;
+  gridbev2egobev(1, 2) = y_start_;
+  gridbev2egobev(2, 2) = 1.f;
+
+  gridbev2egobev(0, 1) = 0.f;
+  gridbev2egobev(1, 0) = 0.f;
+  gridbev2egobev(2, 0) = 0.f;
+  gridbev2egobev(2, 1) = 0.f;
+
+  Eigen::Matrix3f currgrid2adjgrid = gridbev2egobev.inverse() * currEgo2adjEgo_2d * gridbev2egobev;
+
+  CHECK_CUDA(cudaMemcpy(
+    transform_dev, Eigen::Matrix3f(currgrid2adjgrid.transpose()).data(),
+    transform_size_ * sizeof(float), cudaMemcpyHostToDevice));
+}
+
+int BEVDet::doInfer(
+  const camsData & cam_data, std::vector<Box> & out_detections, float & cost_time, int idx)
+{
+  RCLCPP_DEBUG(rclcpp::get_logger("BEVDet"), "---------%d---------", idx + 1);
+
+  auto start = high_resolution_clock::now();
+  CHECK_CUDA(cudaMemcpy(
+    trt_buffer_dev_[buffer_map_["images"]], cam_data.imgs_dev,
+    trt_buffer_sizes_[buffer_map_["images"]], cudaMemcpyDeviceToDevice));
+
+  initCamParams(
+    cam_data.param.cams2ego_rot, cam_data.param.cams2ego_trans, cam_data.param.cams_intrin);
+
+  getAdjBEVFeature(
+    cam_data.param.scene_token, cam_data.param.ego2global_rot, cam_data.param.ego2global_trans);
+
+  if (!trt_context_->executeV2(trt_buffer_dev_)) {
+    RCLCPP_ERROR(rclcpp::get_logger("BEVDet"), "BEVDet forward failing!");
+  }
+
+  adj_frame_ptr_->saveFrameBuffer(
+    trt_buffer_dev_[buffer_map_["curr_bevfeat"]], cam_data.param.scene_token,
+    cam_data.param.ego2global_rot, cam_data.param.ego2global_trans);
+
+  auto end = high_resolution_clock::now();
+
+  postprocess_ptr_->DoPostprocess(post_buffer_, out_detections);
+  CHECK_CUDA(cudaDeviceSynchronize());
+
+  auto post_end = high_resolution_clock::now();
+
+  duration<double> infer_t = post_end - start;
+  duration<double> engine_t = end - start;
+  duration<double> post_t = post_end - end;
+
+  cost_time = infer_t.count() * 1000;
+  RCLCPP_DEBUG(rclcpp::get_logger("BEVDet"), "Inference time: %.5lf ms", engine_t.count() * 1000);
+  RCLCPP_DEBUG(rclcpp::get_logger("BEVDet"), "Postprocess time: %.5lf ms", post_t.count() * 1000);
+  RCLCPP_DEBUG(rclcpp::get_logger("BEVDet"), "Total time: %.5lf ms", infer_t.count() * 1000);
+  RCLCPP_DEBUG(rclcpp::get_logger("BEVDet"), "Detect %ld objects", out_detections.size());
+
+  return EXIT_SUCCESS;
+}
+
+void BEVDet::exportEngine(const std::string & onnxFile, const std::string & trtFile)
```

Comment:
Namespaces: Wrap the function `exportEngine` within the  `autoware_tensorrt_bevdet` namespace.